web/packages/teleterm/src/ui/services/modals/modalsService.ts

```diff
@@ -58,20 +67,44 @@ export class ModalsService extends ImmutableStore<State> {
    * old dialog with the new one.
    * The old dialog is canceled, if possible.
    *
-   * The returned closeDialog function can be used to close the dialog and automatically call the
-   * dialog's onCancel callback (if present).
+   * The passed `abortSignal` or the returned `closeDialog` function can be used to close the dialog
+   * and automatically call the dialog's onCancel callback (if present).
    */
-  openRegularDialog(dialog: Dialog): { closeDialog: () => void } {
+  openRegularDialog(
+    dialog: Dialog,
+    abortSignal?: AbortSignal
+  ): {
+    closeDialog: () => void;
+  } {
+    const onCancelDialog = () => dialog['onCancel']?.();
```

Comment:
We have to document `onCancel` somewhere. I guess in the JSDoc for `openRegularDialog` as we don't have a better place for this. It needs to be explained that `onCancel` can be called with no user interaction within the app if the user launches a deep link while a regular dialog is open. As such, if the dialog presents the user with a decision, `onCancel` should be treated as no decision being made. If possible, the user should be re-prompted to make the decision later on.